subcommands/targets/offline-update.go

```diff
@@ -0,0 +1,279 @@
+package targets
+
+import (
+	"archive/tar"
+	"compress/bzip2"
+	"encoding/json"
+	"fmt"
+	"github.com/foundriesio/fioctl/client"
+	"github.com/foundriesio/fioctl/subcommands"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"io"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	factory string
+	prod    bool
+	tag     string
+)
+
+type (
+	targetInfo struct {
+		version       int
+		ostreeVersion int
+		hardwareID    string
+		buildTag      string
+	}
+)
+
+func init() {
+	offlineUpdateCmd := &cobra.Command{
+		Use:   "offline-update <target-name> <dst> --tag <tag> [--prod]",
+		Short: "Download Target content for an offline update",
+		Run:   doOfflineUpdate,
+		Args:  cobra.ExactArgs(2),
+	}
+	cmd.AddCommand(offlineUpdateCmd)
+	offlineUpdateCmd.Flags().BoolVarP(&prod, "prod", "", false,
+		"Instruct to fetch content of production Target")
+	offlineUpdateCmd.Flags().StringVarP(&tag, "tag", "", "",
+		"Target tag")
+}
+
+func doOfflineUpdate(cmd *cobra.Command, args []string) {
+	factory = viper.GetString("factory")
+	targetName := args[0]
+	dstDir := args[1]
+
+	if len(tag) == 0 {
+		fmt.Fprintln(os.Stderr, "Mandatory flag `tag` is not specified")
+		os.Exit(1)
```

Comment:
i'm not a huge fan of our approach, but we generally error out on bad CLI usage with something like:
```
subcommands.DieNotNil(errors.New("--tag is required"))
```